container_pipeline/workers/build.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+import hashlib
+import json
+import logging
+import os
+
+from container_pipeline.utils import Build, get_job_name
+from container_pipeline.lib.log import load_logger
+from container_pipeline.lib.openshift import Openshift, OpenshiftError
+from container_pipeline.workers.base import BaseWorker
+
+
+class BuildWorker(BaseWorker):
+    """Build worker"""
+
+    def __init__(self, logger=None, sub=None, pub=None):
+        super(BuildWorker, self).__init__(logger, sub, pub)
+        self.openshift = Openshift(logger=self.logger)
+
+    def handle_job(self, job):
+        """Handle job"""
+        parent_build_running = False
+        parents = job.get('depends_on', '').split(',')
+        parents_in_build = []
+        for parent in parents:
+            is_build_running = Build(parent).is_running()
+            if is_build_running:
+                parents_in_build.append(parent)
+            parent_build_running = parent_build_running or \
+                is_build_running
+
+        if parent_build_running:
+            self.logger.info('Parents in build: %s, pushing job: %s back '
+                             'to queue' % (parents_in_build, job))
+            self.queue.put(json.dumps(job), 'master_tube')
+        else:
+            self.logger.info('Starting build for job: %s' % job)
+            success = self.build(job)
+            if success:
+                self.handle_build_success(job)
+            else:
+                self.handle_build_failure(job)
+
+    def build(self, job):
+        """Build job"""
+        namespace = get_job_name(job)
+        project = hashlib.sha224(namespace).hexdigest()
+
+        try:
+            self.openshift.login()
+            build_id = self.openshift.build(project, 'build')
+            if not build_id:
+                return False
+        except OpenshiftError as e:
+            logger.error(e)
+            return False
+
+        Build(namespace).start()
+        build_status = self.openshift.wait_for_build_status(
+            project, build_id, 'Complete')
+        logs = self.openshift.get_build_logs(project, build_id)
```

Comment:
If `build_status` is failed, and there were empty retries, we're still going to be able to fetch build logs from openshift, right? Just checking. Because if not, this and the next two lines till `return build_status` don't serve much purpose. For example if OpenShift itself is down. (I know that's extremely pessimistic of me.) :smile: